scipy/optimize/meson.build

```diff
@@ -0,0 +1,150 @@
+include_dirs = [incdir_numpy, '../_lib/src']
+fortran_ignore_warnings = [
+  '-Wno-tabs', '-Wno-conversion',
+  '-Wno-argument-mismatch', '-Wno-unused-dummy-argument',
+  '-Wno-maybe-uninitialized', '-Wno-unused-label',
+  '-Wno-unused-variable']
+c_ignore_warnings = ['-Wno-unused-variable', 
+                     '-Wno-unused-const-variable']
```

Comment:
I think we can shift this to top level build files and use it in lower level build files for uniformity. For special cases we can write custom flags in lower level build files. What do you say @rgommers ?

Reply:
For the Fortran args I'd say yes, because there's a ton of them and they're often in vendored code that we don't want to fix up. For C I hope we can fix the warnings instead, so I think it'd be preferable not to add these two flags to every extension.